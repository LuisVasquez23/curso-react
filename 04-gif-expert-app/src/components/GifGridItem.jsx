import PropTypes from "prop-types";

const GifGridItem = ({ url, title }) => {
  return (
    <>
      <div className="card">
        <p>{title}</p>
        <img src={url} alt={title} />
      </div>
    </>
  );
};

GifGridItem.prototype = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default GifGridItem;
