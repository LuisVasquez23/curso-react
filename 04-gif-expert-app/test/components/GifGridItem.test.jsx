import { render, screen } from "@testing-library/react";
import GifGridItem from "../../src/components/GifGridItem";

describe("Prueba de GifGridItem", () => {
  const title = "Un tÃ­tulo";
  const url = "https://localhost/algo.jpg";

  test("debe hacer match con el snapshot", () => {
    const { container } = render(<GifGridItem title={title} url={url} />);

    expect(container).toMatchSnapshot();
  });

  test("debe de mostrart la img con el url y el alt inidcado", () => {
    render(<GifGridItem title={title} url={url} />);

    const { src } = screen.getByRole("img");

    expect(src).toBe(url);
  });
});
