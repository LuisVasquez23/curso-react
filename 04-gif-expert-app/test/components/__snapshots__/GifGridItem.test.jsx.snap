// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de GifGridItem debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <p>
      Un título
    </p>
    <img
      alt="Un título"
      src="https://localhost/algo.jpg"
    />
  </div>
</div>
`;
